File: map_python/src/Exercises/Restaurant/01_source_to_stock.py
from src.Restaurant import Ingredient, StockItem, StockState


ingredients = [
    Ingredient("beef", "Beef", 5000, "g", 2.5),
    Ingredient("chicken", "Chicken", 5000, "g", 1.65),
    Ingredient("fish", "Fish", 5000, "g", 1.2),
    Ingredient("potato", "Potato", 800, "g", 0.77),
    Ingredient("tomato", "Tomato", 0, "g", 0.18),
]


stock_items = list(
    map(
        lambda ingredient: StockItem(
            ingredient=ingredient,
            available_quantity=ingredient.quantity,
            state=(
                StockState.EMPTY
                if ingredient.quantity == 0
                else StockState.LOW
                if ingredient.quantity < 1000
                else StockState.AVAILABLE
            ),
        ),
        ingredients,
    )
)


if __name__ == "__main__":
    for stock in stock_items:
        print(stock)

File: map_ts/src/Exercices/Restaurant/01-source-to-stock.ts
import type {
  Ingredient,
  StockItem,
  StockState
} from "../../Restaurant/index.js";


const ingredients: Ingredient[] = [
  {
    id: "beef",
    name: "Beef",
    quantity: 5000,
    unit: "g",
    caloriesPerUnit: 2.5
  },
  {
    id: "chicken",
    name: "Chicken",
    quantity: 5000,
    unit: "g",
    caloriesPerUnit: 1.65
  },
  {
    id: "fish",
    name: "Fish",
    quantity: 5000,
    unit: "g",
    caloriesPerUnit: 1.2
  },
  {
    id: "potato",
    name: "Potato",
    quantity: 800,
    unit: "g",
    caloriesPerUnit: 0.77
  },
  {
    id: "tomato",
    name: "Tomato",
    quantity: 0,
    unit: "g",
    caloriesPerUnit: 0.18
  },
  {
    id: "salad",
    name: "Salad",
    quantity: 0,
    unit: "g",
    caloriesPerUnit: 0.18
  },
  {
    id: "oignon",
    name: "Oignon",
    quantity: 0,
    unit: "g",
    caloriesPerUnit: 0.18
  },
];

const stockItems: StockItem[] = ingredients.map((ingredient) => ({
  ingredient,
  availableQuantity: ingredient.quantity,
  state: ingredient.quantity > 0 ? "available" : "empty"
}));